api/v1alpha2/hierarchy_types.go

```diff
@@ -147,6 +161,15 @@ type HierarchyConfigurationList struct {
 	Items           []HierarchyConfiguration `json:"items"`
 }
 
+// MetaKVP represents a label or annotation
+type MetaKVP struct {
+	// Name is the name of the label or annotation.
+	Name string `json:"name"`
```

Comment:
Yes, this PR doesn't include validation, I'll file issues once it's submitted so that others can pick it up. I mainly wanted to focus on the aspects of the problem that only I could easily implement, but validation is much easier for others to pick up.

I _had_ thought that labels used the term "name" but I re-checked the docs and you're right, they use the term "key." I'll make this change throughout.

Reply:
... we need webhook validation anyway since we need to confirm that the specified labels/annotations are actually managed, not just that they're structurally legal. Fortunately, all the webhooks already exist, we just need to add the logic.